dashboard/config/routes.rb

```diff
@@ -601,6 +601,14 @@
       get 'users/:user_id/contact_details', to: 'users#get_contact_details'
       get 'users/:user_id/school_name', to: 'users#get_school_name'
 
+      patch 'user_school_infos/:id/update_last_confirmation_date', to: 'user_school_infos#update_last_confirmation_date'
+
+      patch 'user_school_infos/:id/update_end_date', to: 'user_school_infos#update_end_date'
+
+      patch 'user_school_infos/:id/update_school_info_id', to: 'user_school_infos#update_school_info_id'
```

Comment:
These single-field `update_*` routes make me think maybe we should be using [resource routing](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default) for user_school_infos instead.  It's more idiomatic to Rails and might be easier to maintain and extend in the future.

In that world, you'd just `PATCH /api/v1/user_school_infos/:id` with the field(s) you wanted to update in the request body.  You've got some special behaviors for different call patterns here so we'd probably have that route set up to only allow very specific request patterns.

Not urgent or required - not even sure this is a good idea.  Might be interesting to bounce off a few folks though.